src/fcntl.rs

```diff
@@ -629,6 +629,125 @@ pub fn flock(fd: RawFd, arg: FlockArg) -> Result<()> {
 
     Errno::result(res).map(drop)
 }
+
+/// Represents valid types for flock.
+///
+/// # Safety
+/// Types implementing this must be `!Clone`.
```

Comment:
To me, the `!` is too small to see clearly.  I think it's better to spell out "not".

```suggestion
/// Types implementing this must not be `Clone`.
```